library/CMakeLists.txt

```diff
@@ -310,20 +324,24 @@ IF(DFHACK_PRERELEASE)
     )
 ENDIF()
 
-# always re-run git-describe if cmake is re-run (e.g. if build ID or version changes)
-EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_SOURCE_DIR}/git-describe.cmake)
-ADD_CUSTOM_TARGET(git-describe
-    COMMAND ${CMAKE_COMMAND}
-        -D dfhack_SOURCE_DIR:STRING=${dfhack_SOURCE_DIR}
-        -D GIT_EXECUTABLE:STRING=${GIT_EXECUTABLE}
-        -D DFHACK_BUILD_ID:STRING=${DFHACK_BUILD_ID}
-        -P ${CMAKE_CURRENT_SOURCE_DIR}/git-describe.cmake
-    COMMENT "Obtaining git commit information"
-)
-ADD_DEPENDENCIES(dfhack-version git-describe)
+configure_file(git-describe.cmake.in ${CMAKE_CURRENT_SOURCE_DIR}/git-describe.cmake @ONLY)
+if (EXISTS ${CMAKE_SOURCE_DIR}/.git/index AND EXISTS ${CMAKE_SOURCE_DIR}/.git/modules/library/xml/index)
+    ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/include/git-describe.h
+      COMMAND ${CMAKE_COMMAND}
+          -D dfhack_SOURCE_DIR:STRING=${dfhack_SOURCE_DIR}
+          -D GIT_EXECUTABLE:STRING=${GIT_EXECUTABLE}
+          -P ${CMAKE_CURRENT_SOURCE_DIR}/git-describe.cmake
+      COMMENT "Obtaining git commit information"
+      DEPENDS ${CMAKE_SOURCE_DIR}/.git/index
+          ${CMAKE_SOURCE_DIR}/.git/modules/library/xml/index
+          ${CMAKE_CURRENT_SOURCE_DIR}/git-describe.cmake
+          include/git-describe.h.in
+    )
+endif ()
 
 ADD_LIBRARY(dfhack SHARED ${PROJECT_SOURCES})
-ADD_DEPENDENCIES(dfhack generate_headers generate_proto_core)
+ADD_DEPENDENCIES(dfhack generate_proto_core)
```

Comment:
Does this actually make a difference?

Reply:
It is something that makes difference for make and is required. I tough I put it back and tested it was working right for make too.

Reply:
Actually. I put it back but to a new line instead of same line. I can switch that back to same line if you want.

Reply:
Right, I was referring to it being split across lines. I don't really care, but `ADD_DEPENDENCIES` suggests that it adds multiple dependencies at once, and I'd expect the order to not matter (but I'm not sure about the `make` backend any more)>

Reply:
Order doesn't matter for dependencies because they must work in any order to allow parallel builds.  I have seen broken build systems where dependency execution order has mattered but then parallel builds where unreliable.